### PR TITLE
Fix and cleanup dependencies

### LINKED ISSUE #163: Problems with dependencies
Technic mod lists `screwdriver` as optional but crashes during loading if `screwdriver` is disabled.
It might be possible to fix this while keeping `screwdriver` optional.

If `moreores` is disabled silver is not available and many recipes are broken.
Moreores is listed as dependency in README.md and should be added to `mod.conf` dependencies too.

### PR DESCRIPTION
**Pull request by OgelGames**
- [x] Remove `depends.txt` files
- [x] Fix #163
- [x] Reduce optional dependencies of `technic_worldgen` using `minetest.register_on_mods_loaded()`
- [ ] ~~Make CNC depend on `technic` (currently optional, which doesn't make sense)~~
- [x] Fix CNC material dependencies.

Not including #214 in this PR because that will be a much bigger change, as opposed these small changes and fixes.

### CODE REVIEW

**Comment by OgelGames** [comment]
Tidied up the commits, so the different changes can be merged as different commits. (rebase merge)

I don't think there is anything else that needs to be done in this PR, so it should be ready for merging (after an approval is added).

File: technic_worldgen/overrides.lua [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+	return desc
+end
+
+local function replace_texture(tile)
+	local new_tile
+	if type(tile) == "string" then
+		new_tile = tile:gsub(
+			"default_steel_block.png",
+			"technic_wrought_iron_block.png"
+		)
```
Comment (S-S-X) [inline technic_worldgen/overrides.lua:42]: Could better be on single line

File: technic_worldgen/overrides.lua [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+	else
+		new_tile = {}
+		for k, v in pairs(tile) do
+			if k == "name" then
+				new_tile[k] = v:gsub(
+					"default_steel_block.png",
+					"technic_wrought_iron_block.png"
+				)
```
Comment (S-S-X) [inline technic_worldgen/overrides.lua:50]: Could better be on single line

**Review by S-S-X — CHANGES_REQUESTED** [review]
Looks good for me, change request just 2 lines for consistent style.

**Review by S-S-X — APPROVED** [review]
Looks good for me 👍 
Also did quick test that registered items (just names) were same in master.

**Pull request merged**